calibrators/intrinsic_camera_calibrator/intrinsic_camera_calibrator/intrinsic_camera_calibrator/calibrators/opencv_calibrator.py

```diff
@@ -31,42 +33,41 @@ class OpenCVCalibrator(Calibrator):
 
     def __init__(self, lock: threading.RLock, cfg: Dict = {}):
         super().__init__(lock, cfg)
-
-        self.radial_distortion_coefficients = Parameter(int, value=2, min_value=0, max_value=6)
+        self.radial_distortion_coefficients = Parameter(int, value=3, min_value=0, max_value=3)
+        self.rational_distortion_coefficients = Parameter(int, value=3, min_value=0, max_value=3)
         self.use_tangential_distortion = Parameter(
             bool, value=True, min_value=False, max_value=True
         )
+        self.enable_prism_model = Parameter(bool, value=True, min_value=False, max_value=True)
         self.fix_principal_point = Parameter(bool, value=False, min_value=False, max_value=True)
         self.fix_aspect_ratio = Parameter(bool, value=False, min_value=False, max_value=True)
 
         self.set_parameters(**cfg)
 
-    def _calibration_impl(self, detections: List[BoardDetection]) -> CameraModel:
-        """Implement the calibrator interface."""
-        flags = 0
-        flags |= cv2.CALIB_FIX_PRINCIPAL_POINT if self.fix_principal_point.value else 0
-        flags |= cv2.CALIB_FIX_ASPECT_RATIO if self.fix_principal_point.value else 0
-        flags |= cv2.CALIB_ZERO_TANGENT_DIST if not self.use_tangential_distortion.value else 0
-        flags |= cv2.CALIB_RATIONAL_MODEL if self.radial_distortion_coefficients.value > 3 else 0
-        flags |= cv2.CALIB_FIX_K6 if self.radial_distortion_coefficients.value < 6 else 0
-        flags |= cv2.CALIB_FIX_K5 if self.radial_distortion_coefficients.value < 5 else 0
-        flags |= cv2.CALIB_FIX_K4 if self.radial_distortion_coefficients.value < 4 else 0
-        flags |= cv2.CALIB_FIX_K3 if self.radial_distortion_coefficients.value < 3 else 0
-        flags |= cv2.CALIB_FIX_K2 if self.radial_distortion_coefficients.value < 2 else 0
-        flags |= cv2.CALIB_FIX_K1 if self.radial_distortion_coefficients.value < 1 else 0
+    def get_model_info(self) -> Tuple[Dict, CameraModelEnum]:
+        return self.get_parameters_values(), CameraModelEnum.OPENCV
 
+    def _calibration_impl(self, detections: List[BoardDetection]) -> OpenCVCameraModel:
+        """Implement the calibrator interface."""
         height = detections[0].get_image_height()
         width = detections[0].get_image_width()
 
-        camera_model = CameraModel()
+        camera_model = make_camera_model(camera_model_type=CameraModelEnum.OPENCV)
```

Comment:
This seems to be a problem from before, but can you check if all `.value` accesses  are done in a locked context?
Applies to the calibration related methods

Reply:
I just checked it and it seems `.value` for all parameters from GUI are reflected here.

Reply:
That is not what I meant.
The parameters are accessed both in the GUI threads and the calibration ones. This raises potential dataraces.
So far I had been making sure (or at least tried) that all accesses to `.value ` were done in a `with self.lock:` context.

I just checked, and this one seems is not. Can you address these cases?
Alternatively, the lock part of ParameterizedClass could be made to be applied to the Parameter class

Reply:
Addressed in https://github.com/tier4/CalibrationTools/pull/208/commits/73eb77bd6fbfb12e67a5707eeddc1082e8fe91d1